tools/src/test/java/org/apache/kafka/tools/TopicCommandIntegrationTest.java

```diff
@@ -85,653 +80,761 @@
+        }
     }
 
-    @ParameterizedTest
-    @ValueSource(strings = {"zk", "kraft"})
-    public void testListTopics(String quorum) {
-        TestUtils.createTopicWithAdmin(adminClient, testTopicName, scalaBrokers, scalaControllers, defaultNumPartitions, defaultReplicationFactor,
-                scala.collection.immutable.Map$.MODULE$.empty(), new Properties()
-        );
+    @ClusterTemplate("generate1")
+    public void testListTopics(TestInfo testInfo) throws InterruptedException {
+        String testTopicName = testInfo.getTestMethod().get().getName() + "-" +
+                TestUtils.randomString(10);
+        try (Admin adminClient = clusterInstance.createAdminClient()) {
+            adminClient.createTopics(Collections.singletonList(new NewTopic(testTopicName, defaultNumPartitions, defaultReplicationFactor)));
+            clusterInstance.waitForTopic(testTopicName, defaultNumPartitions);
 
-        String output = captureListTopicStandardOut(buildTopicCommandOptionsWithBootstrap("--list"));
-        assertTrue(output.contains(testTopicName), "Expected topic name to be present in output: " + output);
+            String output = captureListTopicStandardOut(buildTopicCommandOptionsWithBootstrap("--list"));
+            assertTrue(output.contains(testTopicName), "Expected topic name to be present in output: " + output);
+        }
     }
 
-    @ParameterizedTest
-    @ValueSource(strings = {"zk", "kraft"})
-    public void testListTopicsWithIncludeList(String quorum) {
-        String topic1 = "kafka.testTopic1";
-        String topic2 = "kafka.testTopic2";
-        String topic3 = "oooof.testTopic1";
-        int partition = 2;
-        short replicationFactor = 2;
-        TestUtils.createTopicWithAdmin(adminClient, topic1, scalaBrokers, scalaControllers, partition, replicationFactor,
-                scala.collection.immutable.Map$.MODULE$.empty(), new Properties()
-        );
-        TestUtils.createTopicWithAdmin(adminClient, topic2, scalaBrokers, scalaControllers, partition, replicationFactor,
-                scala.collection.immutable.Map$.MODULE$.empty(), new Properties()
-        );
-        TestUtils.createTopicWithAdmin(adminClient, topic3, scalaBrokers, scalaControllers, partition, replicationFactor,
-                scala.collection.immutable.Map$.MODULE$.empty(), new Properties()
-        );
+    @ClusterTemplate("generate1")
+    public void testListTopicsWithIncludeList() throws InterruptedException {
+        try (Admin adminClient = clusterInstance.createAdminClient()) {
+            String topic1 = "kafka.testTopic1";
+            String topic2 = "kafka.testTopic2";
+            String topic3 = "oooof.testTopic1";
+            int partition = 2;
+            short replicationFactor = 2;
+            adminClient.createTopics(Collections.singletonList(new NewTopic(topic1, partition, replicationFactor)));
```

Comment:
Can we place all these with 
```
  for (String topic: Arrays.asList(topic1, topic2, topic3)) {
                adminClient.createTopics(Collections.singletonList(new NewTopic(topic, partition, replicationFactor)));
                clusterInstance.waitForTopic(topic, partition);
            }
```

Reply:
Also it look like we create topic and wait for it maybe we should have a helper that create topic and wait either in this test or in `clusterInstance`

Reply:
Got it, will fill another PR.

Reply:
File another PR, https://github.com/apache/kafka/pull/16852, PTAL.